yyjson/document.c

```diff
@@ -22,17 +22,13 @@ static PyObject *element_to_primitive(yyjson_val *val, bool raw_as_decimal);
 static PyObject *pathlib = NULL;
 static PyObject *path = NULL;
 
-/**
- * Count the number of UTF-8 characters in the given string.
- */
-static inline size_t num_utf8_chars(const char *src, size_t len) {
-  size_t count = 0;
+static inline bool is_ascii(const char * src, size_t len) {
   for (size_t i = 0; i < len; i++) {
-    if (yyjson_likely(src[i] >> 6 != 2)) {
-      count++;
+    if ((src[i] & 0x80) != 0) {
```

Comment:
technically these implement different logic. Are we certain we're correct here? Would there be any benefit to using their `yyjson_likely()` macro for code layout optimization?

Reply:
It's the same check but I thought the AND might be faster than the shift since we'd have to mask and shift `((src[i] & 0xff) >> 6) != 2` to fix the bug that was there before. I didn't think the hint (`yyjson_unlikely()`) would help any in this case since we're potentially iterating through the whole loop anyway but i can add it back if you think it's worth it. The function is so small that there's probably not much more optimization to be had.